### DISCUSSION TITLE
UWG morphometric calculator (grid)

### DISCUSSION THREAD

**Discussion by aleksandraradecka1** [UMEP-dev/UMEP#445] [Q&A]
Hello Colleagues,

I discovered that because of flat parts of my research area, there are problems with using UWG processor. To put it more precisely, the values of morphometric indices (pai, fai, zH, zHmax, zHstd, zd, z0 i wai) are for some grids equal to zero or close to it. Do you maybe know what are the minimal values for these indices which I could use for these grids to make the algorithm run?

I would be greatful for the help.
Best regards, 
Aleksandra

**Comment by biglimp** [comment]
All the mentioned parameters can be zero. Well maybe not zd and z0. If you e.g. have no objects (buildings), the first 5 will be zero. If uwg produce errors then maybe we should adjust these to a very small number, so that the model does not break. I have not tested this, i.e. what number that are appropriate. You can also submit an issue to the uwg github page and tell them about this issue. The best thing would be if they solved it at thier end. /Fredrik

**Reply by aleksandraradecka1** [reply]
Thanks for the reply. 

I have made a few tests in the last days and it turns out that the algorithm requires buildings' height to be at least 2 m, otherwise it gives "math error". I did not manage to achieve this through morphological calculator's (grid) file, it only started working when I changed "bldHeight" to 2.0 in the UWG prepare's output file (therefore one step later). 

I however feel that this workaround is not perfect - in the next step (in UWG Analyzer) I started to have a numpy error :( 

I will post the problem to the Authors.

**Reply by biglimp** [reply]
Ok, we should think how we can solve this the easiest way. Maybe we should force zH to be 2 meter then.
Report any error in the UWG Analyzer alos so that we can fix them.

**Reply by aleksandraradecka1** [reply]
I have just checked - yes, 'zH' is copied directly as 'bldHeight'. This value previously did not work for me, and now I think I see why - 'verToHor' also needs a specific value e.g. 1.0 (for sure it may also be some other value, but not 0.001 it turns out). 

The error in UWG Analyzer looks like this:
<img width="860" alt="error_analyzer" src="https://user-images.githubusercontent.com/44131961/184871779-07381824-4a3f-474c-8d7a-7cc38872a87f.png">

**Reply by biglimp** [reply]
Can you please attach your data so that I can explore further.

**Reply by aleksandraradecka1** [reply]
Of course. Here they are: https://drive.google.com/file/d/1ZkG7_Jf-tWOuise3lNPMiz4v4oc1aHQm/view?usp=sharing

Please let me know if sth is missing.

**Reply by biglimp** [reply]
hm, I see that your epw-file is not a full year. Are you able to run the UWG-model with an epw-file that is just a part of a full year? I think the setup in UMEP requires a full year of data. By saying that, you can still model a small fraction of that year by defining startdate and no of days to model in the tool.

**Reply by aleksandraradecka1** [reply]
Yes, it is true. I wrote a little note on this in the "README" file, but I have placed it inside the folder so you probably did not see it. You need to select some days in January or Febrary in order to make the tool work in QGIS. Unfortunately I do not have a file with full year right now.

**Reply by biglimp** [reply]
You can just fill your epw-file.with dummy data as long as you habe good data during the days you want to model.

**Reply by aleksandraradecka1** [reply]
Yes, this is what I will do for now. I hope later my colleague will prepare data for the whole year to give te analysis a bit more flexibility.
